packages/devtools_app/lib/src/screens/vm_developer/vm_library_display.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../shared/theme.dart';
+import 'vm_developer_common_widgets.dart';
+import 'vm_object_model.dart';
+
+/// A widget for the object inspector historyViewport displaying information
+/// related to library objects in the Dart VM.
+class VmLibraryDisplay extends StatelessWidget {
+  const VmLibraryDisplay({
+    required this.library,
+  });
+
+  final LibraryObject library;
+
+  @override
+  Widget build(BuildContext context) {
+    final dependencies = library.obj.dependencies;
+    return VmObjectDisplayBasicLayout(
+      object: library,
+      generalDataRows: _libraryDataRows(library),
+      expandableWidgets: [
+        if (dependencies != null)
+          LibraryDependencies(dependencies: dependencies)
+      ],
+    );
+  }
+
+  /// Generates a list of key-value pairs (map entries) containing the general
+  /// information of the library object [library].
+  List<MapEntry<String, WidgetBuilder>> _libraryDataRows(
+    LibraryObject library,
+  ) {
+    return [
+      ...vmObjectGeneralDataRows(library),
+      selectableTextBuilderMapEntry(
+        'URI',
+        (library.obj.uri?.isEmpty ?? false)
+            ? library.script?.uri
+            : library.obj.uri,
+      ),
+      selectableTextBuilderMapEntry(
+        'VM Name',
+        library.vmName,
+      ),
+    ];
+  }
+}
+
+// An expandable tile displaying a list of library dependencies
+class LibraryDependencies extends StatelessWidget {
+  const LibraryDependencies({
+    required this.dependencies,
+  });
+
+  final List<LibraryDependency> dependencies;
+
+  List<Row> dependencyRows(BuildContext context) {
+    return <Row>[
+      for (final dep in dependencies)
+        Row(
+          children: [
+            Flexible(
+              child: SelectableText(
+                _dependencyDescription(dep),
+                style: Theme.of(context).fixedFontStyle,
+              ),
+            ),
+          ],
+        )
+    ];
+  }
+
+  String _dependencyDescription(LibraryDependency dependency) {
```

Comment:
make this public (can be `@visibleForTesting`) and add a test for it

Reply:
I've made the tests for the LibraryDependencies widget overall. If I only test  `dependencyDescription` then I would have to make an empty LibraryDependencies widget `final widget = LibraryDependencies(dependencies: [])` and then call `widget.dependencyDescription(dep...)`. 
Would you like me to change to this way or is it okay to keep it private and test for the LibraryDependencies widget?

Reply:
Ah I see, I didn't realize this was in a widget class at first. A better option would be to make this an extension method instead that you can easily test. So in this file you'd write

```dart
extension LibraryDependencyExtension on LibraryDependency {
  String get description {
    ...
  }
}
```